src/SFA.DAS.EmployerFinance.Web.UnitTests/Orchestrators/WhenGettingCoursePaymentSummary.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoFixture;
+using FluentAssertions;
+using MediatR;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Common.Domain.Types;
+using SFA.DAS.EAS.Account.Api.Client;
+using SFA.DAS.EmployerFinance.Interfaces;
+using SFA.DAS.EmployerFinance.Models.Account;
+using SFA.DAS.EmployerFinance.Models.Payments;
+using SFA.DAS.EmployerFinance.Models.Transaction;
+using SFA.DAS.EmployerFinance.Queries.FindAccountCoursePayments;
+using SFA.DAS.EmployerFinance.Queries.GetEmployerAccount;
+using SFA.DAS.EmployerFinance.Web.Orchestrators;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EmployerFinance.Web.UnitTests.Orchestrators
+{
+    [Parallelizable]
+    public class WhenGettingCoursePaymentSummary
+    {
+        private IFixture _fixture = new Fixture();
+        private EmployerAccountTransactionsOrchestrator _sut;
+        private Mock<IAccountApiClient> _accountApiMock;
+        private Mock<IMediator> _mediatorMock;
+        private Mock<ICurrentDateTime> _currentTimeMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mediatorMock = new Mock<IMediator>();
+            _accountApiMock = new Mock<IAccountApiClient>();
+            _currentTimeMock = new Mock<ICurrentDateTime>();
+
+            _mediatorMock
+                .Setup(mock => mock.SendAsync(It.IsAny<GetEmployerAccountHashedQuery>()))
+                .ReturnsAsync(new GetEmployerAccountResponse
+                {
+                    Account = new Account { ApprenticeshipEmployerType = ApprenticeshipEmployerType.NonLevy }
+                });
+
+            SetupGetCoursePaymentsResponse(2019, 9);
+
+            _sut = new EmployerAccountTransactionsOrchestrator(
+                _accountApiMock.Object, 
+                _mediatorMock.Object, 
+                _currentTimeMock.Object,
+                Mock.Of<ILog>());
+        }
+
+        [Test]
+        [TestCase(1, Description = "Course payment summary for single apprentice")]
+        [TestCase(9, Description = "Course payment summary for multiple apprentice")]
+        public async Task ThenSummariesForEachApprenticeShouldBeCreated(int numberOfApprentices)
+        {
+            // Arrange
+            var coursePayments = CreateCoursePayments(numberOfApprentices, 1);
+            SetupGetCoursePaymentsResponse(2019, 9, coursePayments);
+
+            // Act
+            var response = await _sut.GetCoursePaymentSummary("abc123", 888888, "A course", 4, null, new DateTime(2019, 9, 1), new DateTime(2019, 9, 30), "userId");
```

Comment:
Will the hard coded dates in the tests cause test failures when the year changes?

Reply:
No issue with the dates, this is getting the summary for a timespan so doesn't matter